src/pudl/analysis/record_linkage/cross_year.py

```diff
@@ -0,0 +1,246 @@
+"""Scikit-Learn classification pipeline for linking records between years.
+
+FERC does not publish plant IDs, so to be able to do any time-series analysis, it is
+required to assign IDs ourselves. The pipeline has been generalized to work with any
+inter-year record linkage problem.
+"""
+from typing import Any, Literal
+
+import numpy as np
+import pandas as pd
+from pydantic import BaseModel
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.cluster import AgglomerativeClustering
+from sklearn.compose import ColumnTransformer
+from sklearn.decomposition import PCA
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics import pairwise_distances
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import MinMaxScaler, OneHotEncoder
+
+import pudl
+from pudl.analysis.record_linkage.cleaning_steps import CleaningRules
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+_TFIDF_DEFAULT_OPTIONS = {
+    "analyzer": "char",
+    "ngram_range": (2, 10),
+}
+
+_ONE_HOT_DEFAULT_OPTIONS = {
+    "categories": "auto",
+}
+
+_CLEANING_FUNCTIONS = {
+    "null_to_zero": lambda df, cols: df[cols].fillna(value=0.0),
+    "null_to_empty_str": lambda df, cols: df[cols].fillna(value=""),
+    "fix_int_na": lambda df, cols: pudl.helpers.fix_int_na(df, columns=cols)[cols],
+}
+
+
+class ColumnTransform(BaseModel):
+    """Configuration for a single column transform in the CrossYearLinker.
+
+    Defines a scikit-learn transformer to be used by a ColumnTransformer. See
+    https://scikit-learn.org/stable/modules/generated/sklearn.compose.ColumnTransformer.html#sklearn.compose.ColumnTransformer
+    for more information.  The full set of column transforms will transform the
+    input DataFrame into a feature matrix that will be to cluster records.
+
+    The 'transformer' should be a string or an initialized scikit-learn BaseEstimator.
+    If it is a string, it should be one of the following:
+        'string' - Applies a TfidfVectorizer to the column.
+        'category' - Applies a OneHotEncoder to the column.
+        'number' - Applies a MinMaxScaler to the column.
```

Comment:
I suspect this indentation is what's causing the docs build to fail.

Reply:
Yeah I think you're right, should be fixed now.